### PR TITLE
Update imagestream backup plugin to copy images to migration registry

### PR DESCRIPTION
**Pull request by sseago**
Update imagestream backup plugin to copy images to migration registry
    
    imagestream backup plugin uses containers/image to copy locally-managed
    images from the internal registry to the s3-backed migration registry.
    
    At this point, the plugin assumes that:
    1) internal registry is running insecure (http), but requires auth
    2) velero user token has privileges to access internal registry images
    3) migration registry is both insecure and requires no credentials

The second commit includes the backup.go changes which implement the PR functionality.
The first commit includes all of the required deps in vendor/ and some Makefile/Gopkg changes required for them.

### CODE REVIEW

File: velero-plugins/imagestream/backup.go
Comment (dymurray) [inline velero-plugins/imagestream/backup.go:219]: Do you want this function exported? Or will the reuse of this be in the same package?

**Review by dymurray — APPROVED** [review]
LGTM. Makefile changes look sane as well until we can get your upstream PR merged.

**Comment by sseago** [comment]
My containers/storage upstream PR has been closed in favor of a different fix by the upstream maintainer. I've updated the Makefile to apply a patch based on that PR instead.

File: velero-plugins/imagestream/backup.go
Comment (sseago) [reply velero-plugins/imagestream/backup.go:219]: For now, it should only be needed within the package.

**Pull request merged**
